### PR TITLE
feat: use the enabled field from `MarketMap`

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-1089

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -168,8 +168,11 @@ build-configs:
 	@jq '.consensus["params"]["abci"]["vote_extensions_enable_height"] = "2"' $(GENESIS) > $(GENESIS_TMP) && mv $(GENESIS_TMP) $(GENESIS)
 	@jq '.app_state["oracle"]["currency_pair_genesis"] += [{"currency_pair": {"Base": "BTC", "Quote": "USD"},"currency_pair_price": null,"nonce": "0"}]' $(GENESIS) > $(GENESIS_TMP) && mv $(GENESIS_TMP) $(GENESIS)
 	@jq '.app_state["oracle"]["next_id"] = "2"' $(GENESIS) > $(GENESIS_TMP) && mv $(GENESIS_TMP) $(GENESIS)
+	@dasel put -r toml 'rpc.laddr' -f $(CONFIG_TOML) -t string -v "tcp://0.0.0.0:26657"
 	@dasel put -r toml 'telemetry.enabled' -f $(APP_TOML) -t bool -v true
-	@dasel put -r toml 'instrumentation.enabled' -f $(CONFIG_TOML) -t bool -v true
+	@dasel put -r toml 'api.enable' -f $(APP_TOML) -t bool -v true
+	@dasel put -r toml 'grpc.address' -f $(APP_TOML) -t string -v "0.0.0.0:9090"
+	@dasel put -r toml 'api.address' -f $(APP_TOML) -t string -v "tcp://0.0.0.0:1317"
```
Comment (aljo242) [inline Makefile:175]: this allows us to curl query the node

File: config/dydx/market.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
           "Quote": "USD"
         },
         "decimals": 8,
-        "min_provider_count": 3
+        "min_provider_count": 3,
+        "enabled": true
```
Comment (aljo242) [inline config/dydx/market.json:11]: enable everything

File: oracle/types/market.go [hunk 3]
```diff
@@ -28,6 +24,10 @@ func ProviderTickersFromMarketMap(
 	// Iterate through every single market and its provider configurations to find the
 	// provider configurations that match the provider name.
 	for _, market := range marketMap.Markets {
+		if !market.Ticker.Enabled {
```
Comment (aljo242) [inline oracle/types/market.go:27]: this is actually where we skip the disabled providers

File: oracle/types/market_test.go [hunk 4]
```diff
@@ -26,23 +26,32 @@ func TestProviderTickersFromMarketMap(t *testing.T) {
 			err:      false,
 		},
 		{
-			name:     "invalid market map",
+			name:     "single disabled market",
```
Comment (Eric-Warehime) [inline oracle/types/market_test.go:29]: I think the thing which is missing here for me is:
* what do we do if there is a transitive dependency on a disabled market?
For example, do we allow `USDT/USDC` to be disabled if that's our normalize market for other things?

File: providers/apis/dydx/parse.go [hunk 5]
```diff
@@ -115,6 +115,7 @@ func (h *APIHandler) CreateTickerFromMarket(market dydxtypes.MarketParam) (mmtyp
 		CurrencyPair:     cp,
 		Decimals:         uint64(market.Exponent * -1),
 		MinProviderCount: uint64(market.MinExchanges),
+		Enabled:          true,
```
Comment (aljo242) [inline providers/apis/dydx/parse.go:118]: We always enable everything parsed from dydx

File: oracle/types/market_test.go (hunk 4 above)
Comment (davidterpay) [reply oracle/types/market_test.go:29]: I was thinking the same thing, if a given feed should be accessible but the overall market shouldn't be calculated. Is this a case we should be worried about?

File: Makefile (hunk 1 above)
Comment (davidterpay) [reply Makefile:175]: what is this doing?

**Comment by davidterpay** [comment]
Adding a preview for this to see how it runs

File: Makefile (hunk 1 above)
Comment (Eric-Warehime) [reply Makefile:175]: I think the sdk defaults the node's API endpoint to off unless you set `api.enable` and `api.address` I guess is the port it's serving on. Not sure what the default value for that is--thought it was `localhost:1317` which effectively is the same?

File: oracle/types/market.go [hunk 6]
```diff
@@ -13,10 +13,6 @@ func ProviderTickersFromMarketMap(
 	name string,
 	marketMap mmtypes.MarketMap,
 ) ([]ProviderTicker, error) {
-	if err := marketMap.ValidateBasic(); err != nil {
```
Comment (Eric-Warehime) [inline oracle/types/market.go:16]: Where does this get run now?

Comment (aljo242) [reply oracle/types/market.go:16]: This is always already run when the market map is read in

Comment (aljo242) [reply oracle/types/market.go:16]: https://github.com/skip-mev/slinky/blob/02f198691eadeb1cb1289d272989823e4c1549cc/x/marketmap/types/utils.go#L10

when read in from file

Comment (aljo242) [reply oracle/types/market.go:16]: https://github.com/skip-mev/slinky/blob/02f198691eadeb1cb1289d272989823e4c1549cc/providers/apis/marketmap/api_handler.go#L93

when read by provider

**Review by Eric-Warehime — APPROVED** [review]

File: providers/apis/dydx/multi_market_map_fetcher_test.go [hunk 7]
```diff
@@ -96,6 +96,7 @@ func TestDYDXMultiMarketMapFetcher(t *testing.T) {
 									CurrencyPair:     slinkytypes.NewCurrencyPair("BTC", "USD"),
 									Decimals:         8,
 									MinProviderCount: 1,
+									Enabled:          true,
```
Comment (davidterpay) [inline providers/apis/dydx/multi_market_map_fetcher_test.go:99]: Does the test case below also have to have it be enabled?

Comment (aljo242) [reply providers/apis/dydx/multi_market_map_fetcher_test.go:99]: yeah, it should I guess - we are only checking that a field exists under that key.

not any of the fields themselves like `Enabled`

**Review by davidterpay — APPROVED** [review]

**Pull request merged**